app/forms/select_email_form.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+class SelectEmailForm
+  include ActiveModel::Model
+  include ActionView::Helpers::TranslationHelper
+
+  attr_reader :user, :selected_email
+
+  validate :validate_owns_selected_email
+
+  def initialize(user)
+    @user = user
+  end
+
+  def submit(params)
+    @selected_email = params[:selection]
+
+    success = valid?
+    process_successful_submission if success
+    FormResponse.new(success:, errors:)
+  end
+
+  private
+
+  def process_successful_submission
+    EmailAddress.update_last_sign_in_at_on_user_id_and_email(
+      user_id: @user.id,
+      email: EmailAddress.find(@selected_email).email,
+    )
```

Comment:
Following from my [previous comment](https://github.com/18F/identity-idp/pull/10951#discussion_r1697563808), why do we still need to update the last sign-in? If historically we used last sign-in to indicate the actual email/password submission, I don't know that the assumptions around it necessarily extend to selecting an email to share with a partner.

Reply:
It appeared that the completions controller gets the email that goes into that final passthrough to the SP from the last email.
https://github.com/18F/identity-idp/blob/main/app/services/displayable_pii_formatter.rb#L39 by way of the completions presenter https://github.com/18F/identity-idp/blob/main/app/presenters/completions_presenter.rb#L130
I don't know if my logic adds up but without changing the last sign-in email the shared value doesn't go through.

Reply:
Should we update the logic of that method to consider the session value if it exists?

Reply:
I have revised how the email gets passed to the SP without altering the last sign in.